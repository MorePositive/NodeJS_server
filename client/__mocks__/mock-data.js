export const mockItem = [{
  firstName: "Ivan",
  lastName: "Ivanov",
  phone: 50887562,
  email: "asd@ads.com"
}];

export const mockItems = [{
  firstName: "Oleg",
  lastName: "Olegov",
  phone: 68333121,
  email: "mail@test.com"
},
{
  firstName: "John",
  lastName: "Smith",
  phone: 67125748,
  email: "test@mail.com"
}];

export const mockTitle = "sunt aut facere repellat provident occaecati excepturi optio reprehenderit";

export const mockBody = "quia et suscipit suscipit recusandae consequuntur expedita et cum reprehenderit molestiae ut ut quas totam nostrum rerum est autem sunt rem eveniet architecto";
