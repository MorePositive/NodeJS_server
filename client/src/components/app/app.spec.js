import { shallow } from 'enzyme';
import React from 'react';

import { mockTitle, mockBody } from '../../../__mocks__/mock-data'
import App from './app';

const mockJsonPromise = Promise.resolve({ userId: 1, id: 1, title: mockTitle, body: mockBody });
const mockFetchPromise = Promise.resolve({ json: () => mockJsonPromise });
global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);
const event = { preventDefault: () => {} };

describe("App component", () => {
  let component;
  let instance;

  beforeEach(() => {
    component = shallow(<App />);
    instance = component.instance();
  });

  it("should render App component", () => {
    const wrapper = component.find(".App");
    expect(wrapper).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });

  it("getData: should make response to api", () => {
    instance.getPost(event);
    expect(global.fetch).toHaveBeenCalled();
  });

  it("getData: should make response with exact query", () => {
    instance.getPost(event);
    expect(global.fetch).toHaveBeenCalledWith("https://jsonplaceholder.typicode.com/posts/1");
  });

  it("setData: should add an item to local state", () => {
    const prevLength = component.state().data.length;
    instance.setState({ data : [{ email: "test@test.com" }] });
    expect(component.state().data.length).toBe(prevLength + 1);
  });

  it("createItem: should create an item", () => {
    const result = instance.createItem("Ivan", "Ivanov", 50887562, "asd@ads.com");
    expect(result.firstName).toBe("Ivan");
    expect(result.lastName).toBe("Ivanov");
    expect(result.phone).toBe(50887562);
    expect(result.email).toBe("asd@ads.com");
  });

  it("addItem: should transport data to setData method and refresh the state", () => {
    instance.addItem("first", "last", 888888, "mail@mail.com");
    const [ stateData ] = component.state().data;
    expect(stateData.firstName).toBe("first");
    expect(stateData.lastName).toBe("last");
    expect(stateData.phone).toBe(888888);
    expect(stateData.email).toBe("mail@mail.com");
  });

  it("clearData: should clear state data", () => {
    instance.setState({ data: [{ test: "test"}] });
    expect(component.state().data.length).toEqual(1);
    instance.clearData();
    expect(component.state().data.length).toEqual(0);
  });
});
