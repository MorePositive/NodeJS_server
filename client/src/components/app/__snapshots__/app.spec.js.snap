// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render App component 1`] = `
<div
  className="App"
>
  <Form
    clearData={[Function]}
    getPost={[Function]}
    onItemAdded={[Function]}
  />
  <ItemList
    data={Array []}
  />
</div>
`;
